controllers/front/OrderController.php

```diff
@@ -150,7 +151,12 @@ protected function bootstrap()
     protected function saveDataToPersist(CheckoutProcess $process)
     {
         $data = $process->getDataToPersist();
-        $data['checksum'] = $this->cartChecksum->generateChecksum($this->context->cart);
+        
+        try {
+            $data['checksum'] = $this->cartChecksum->generateChecksum($this->context->cart);
```

Comment:
I don't think it should not be up to CartChecksum to verify the integrity of the Cart, because it violates the Single Responsibility Principle.

You should create a `CartValidator` that does this job, with a method `bool isValid($cart)`, and eventually `int[] getInvalidAdresses()` which returns an array of invalid address ids. No need to be throwing exceptions around anymore.